text/0000-build-systems.md

```diff
@@ -0,0 +1,446 @@
+much easier than it is today.
+
+After extensive discussion with stakeholders, there appear to be two distinct
+kinds of use-cases (or "customers") involved here:
+
+- **Mixed build systems**, where building already involves a variety of
+  language- or proeject-specific build systems. For this use case, the desire is
+  to use Cargo as-is, except for some specific concerns. Those concerns take a
+  variety of shapes: customizing caching, having a local crate registry, custom
+  handling for native dependencies, and so on. Addressing these concerns well
+  means adding new points of extensibility or control to Cargo.
+
+- **Homogenous build systems** like [Bazel], where there is a single prevailing
+  build system and methodology that works across languages and projects and is
+  expected to drive all aspects of the build. In such cases the goal of Cargo
+  integration is largely *interoperability*, including easy use of the crates.io
+  ecosystem and Rust-centric tooling, both of which expect Cargo-driven build
+  management.
+
+[Bazel]: https://bazel.build/
+
+The interoperability constraints are, in actuality, hard constraints around
+*any* kind of integration.
+
+In more detail, a build system integration *must*:
+
+- Make it easy for the outer build system to control the aspects of building
+  that are under its purview (e.g. artifact management, caching, network access).
+- Make it easy to depend on arbitrary crates in the crates.io ecosystem.
+- Make it easy to use Rust tooling like `rustfmt` or the RLS with projects that
+  depend on the external build system.
+
+A build system integration *should*:
+
+- Provide Cargo-based or Cargo-like workflows when developing Rust projects, so
+  that documentation and guidance from the Rust community applies even when
+  working within a different build system.
+- To the extent possible, support Cargo concepts in a smooth, first-class way in
+  the external build system (e.g. Cargo features, profiles, etc)
+
+This RFC does not attempt to provide a detailed solution for all of the needed
+extensibility points in Cargo, but rather to outline a general plan for how to
+get there over time. Individual components that add significant features to
+Cargo will need follow-up RFCs before stabilization.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The plan proposed in this RFC is to address the two use-cases from the
+[motivation] section in parallel:
+
+- **For the mixed build system case**, we will triage feature requests and work on
+  adding further points of extensibility to Cargo based on expected impact. Each
+  added point of extensibility should ease build system integration for another
+  round of customers.
+
+- **For the homoegenous build system case**, we will immediately pursue
+  extensibility points that will enable the external build system to perform
+  many of the tasks that Cargo does today--but while still meeting our
```

Comment:
What sort of timeline are you looking at for "immediately pursu[ing] extensibility points"?